app/src/main/java/org/mozilla/tv/firefox/session/SessionRepo.kt

```diff
@@ -34,9 +38,16 @@ class SessionRepo(
         val currentUrl: String
     )
 
+    enum class Event {
+        YouTubeBack, ExitYouTube
+    }
+
     private val _state = MutableLiveData<State>()
     val state: LiveData<State> = _state
 
+    private val _events: Subject<Event> = PublishSubject.create()
+    val events = _events.hide()
```

Comment:
Okay, so you `hide()` here to prevent others from calling operations like `next()` on `_events`? What's the scope of that restriction, e.g. this class, and is it redundant because `_events` is private to the class anyway? (I guess it seems like good hygiene in any case.)
And no one can just call `events.onNext` because it's just an Observable, not an ObservableSource.

Reply:
Code local to the class can still push values through `_events`, but anything else only has access to the read-only `events`.  This is similar to what we were doing before (with `val events: LiveData<T> = _events`, see `SessionRepo.events`), but with the upside that `Observables` obtained using `.hide()` cannott be cast back to subjects (`LiveData` can be cast back and forth freely at any time).